File: engine/julia/vix_term_structure_engine.jl
# Module AG5 (Julia): VIX Term Structure, Futures Roll Yield & Volatility ETP Arbitrage Engine
# Mathematical modeling of VIX futures curves, roll yields, and VVIX volatility-of-volatility metrics.

module VixTermStructureEngine

export analyze_vix_curve, evaluate_vvix_risk

function analyze_vix_curve(spot_vix::Float64, m1::Float64, m2::Float64, delta_days::Int)
    slope = m2 - m1
    d = max(1, delta_days)
    roll_yield = ((m2 - m1) / m1) * (365.0 / d) * 100.0
    regime = slope > 0.15 ? "CONTANGO" : (slope < -0.15 ? "BACKWARDATION" : "FLAT")
    
    return (
        slope = slope,
        roll_yield = roll_yield,
        regime = regime
    )
end

function evaluate_vvix_risk(spot_vix::Float64, vvix::Float64)
    is_elevated = vvix >= 115.0
    action = is_elevated ? "BUY_VIX_CALL_SPREADS" : "STANDARD_HARVEST"
    return (is_elevated = is_elevated, action = action)
end

end

File: engine/rust/src/vix_term_structure_engine.rs
//! Module AG4 (Rust): VIX Term Structure, Futures Roll Yield & Volatility ETP Arbitrage Engine.
//! Provides SIMD-optimized evaluation of VIX term slope, roll yield, and VVIX spike alerts.

#[repr(C, align(64))]
#[derive(Debug, Clone, Copy)]
pub struct VixTermStructureState {
    pub spot_vix: f64,
    pub m1_futures_price: f64,
    pub m2_futures_price: f64,
    pub term_slope: f64,
    pub annualized_roll_yield: f64,
    pub vvix_index: f64,
    pub contango_flag: u32,
    pub tail_risk_spike_flag: u32,
    pub _padding: [u8; 8],
}

impl VixTermStructureState {
    pub fn new(spot_vix: f64, m1: f64, m2: f64, vvix: f64, delta_days: i32) -> Self {
        let slope = m2 - m1;
        let d = delta_days.max(1) as f64;
        let roll_yield = ((m2 - m1) / m1) * (365.0 / d) * 100.0;
        let contango = if slope > 0.15 { 1 } else { 0 };
        let tail_risk = if vvix >= 115.0 { 1 } else { 0 };

        Self {
            spot_vix,
            m1_futures_price: m1,
            m2_futures_price: m2,
            term_slope: slope,
            annualized_roll_yield: roll_yield,
            vvix_index: vvix,
            contango_flag: contango,
            tail_risk_spike_flag: tail_risk,
            _padding: [0u8; 8],
        }
    }
}
